### PR TITLE
Add check for social_auth table on user creation

### PR DESCRIPTION
**Pull request by Zlopez**
Fixes https://github.com/release-monitoring/anitya/issues/565

### CODE REVIEW

File: anitya/tests/base.py [hunk 1]
```diff
@@ -287,5 +288,23 @@ def create_flagged_project(session):
     return flag
 
 
+def create_user(session):
```
Comment (jeremycline) [inline anitya/tests/base.py]: With tests I think explicit is preferable to DRY. Having a function to create a user is indirection that makes the test harder to read an understand at a glance. I don't think this saves a huge amount of typing and isn't worth that indirection.

Comment (Zlopez) [reply anitya/tests/base.py]: I'm not sure if this is true, the user is created on many places in tests. When I added this feature, almost 50 tests started failing. So this saved me a lot typing.

File: anitya/db/events.py [hunk 2]
```diff
@@ -57,3 +58,23 @@ def set_ecosystem(session, flush_context, instances):
...
+        session (sqlalchemy.orm.session.Session): The session that is about to be committed.
+        flush_context (sqlalchemy.orm.session.UOWTransaction): Unused.
+        instances (object): deprecated and unused
+
+    Raises:
+        ValueError: If the social_auth table is not associated with this user.
+    """
+    for new_obj in session.new:
+        if isinstance(new_obj, User):
+            if len(new_obj.social_auth.all()) == 0:
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply anitya/db/events.py]: Good point

File: anitya/app.py [hunk 3]
```diff
@@ -143,10 +143,16 @@ def integrity_error_handler(error):
...
         Session.rollback()
-        other_user = models.User.query.filter_by(email=error.params['email']).one()
-        social_auth_user = other_user.social_auth.filter_by(user_id=other_user.id).one()
-        msg = ("Error: There's already an account associated with your email, "
-               "authenticate with {}.".format(social_auth_user.provider))
-        return msg, 400
+        if 'social_auth' in error.params:
+            msg = (
+                "Error: Authentication with authentication provider failed. "
+                "Please try again later...")
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply anitya/app.py]: This means, that the user is already created, but without social_auth table entry. This is why there is this error message. I was not sure what HTTP error should be used for this. It should work when tried later.

File: anitya/tests/base.py (hunk 1 above)
Comment (Zlopez) [reply anitya/tests/base.py]: You are probably right in this, it is better if this is done in the same place as tests, not in imported code.

**Comment by Zlopez** [comment]
I did a rebase to split this PR to fix and test only PR (https://github.com/release-monitoring/anitya/pull/619).

**Comment by Zlopez** [comment]
I did a rebase after merge of #619. Tests are passing now.

**Comment by Zlopez** [comment]
Resolved conflict.

**Comment by Zlopez** [comment]
@jeremycline is this looking good to you now?

**Comment by Zlopez** [comment]
Because @jeremycline approved this one previously and only thing I did was to fix failing tests which are now going through. I will merge this manually.

**Pull request merged**